test/python_test.rb

```diff
@@ -54,6 +54,12 @@ module Rock
             python_bin, version = Rock.find_python(ws: @ws, version: "<100.0" )
             assert(File.exist?(python_bin))
             assert(version =~ /[0-9]+\.[0-9]+/)
+
+            # Python3 has higher priority, so should be picked
+            python_path = `which python3`.strip()
+            if File.exists?(python_path)
+                assert(version =~ /3.[0-9]+/)
```

Comment:
The current implementation ensures that autodetection of python prefers python3, since it is currently the only version supported. So it is picked (a) if available and (b) no other version is requested.
(b) means, that python2 and other versions can still be selected by the user.